### ISSUE TITLE
Reactivity lost when using Meteor.call() to change document

### ISSUE THREAD

**Issue by nadeemja** [cult-of-coders/redis-oplog#119]

**Comment by theodorDiaconu** [comment]
@nadeemja pls tell us how does your call look like, and how your publication looks like.

**Comment by nadeemja** [comment]
__Method__
```javascript
    const _this = this;

    Meteor.methods({
      likeDocument(collectionName, documentId) {
        check(collectionName, String);
        check(documentId, String);

        const Collection = _this[collectionName];

        const document = Collection.findOne({ _id: documentId });

        if (!document) {
          return false;
        }

        const userAlreadyLiked = _.contains(document.likedBy, Meteor.userId());

        if (userAlreadyLiked) {
          Collection.update({
            _id: documentId,
          }, {
            $pull: {
              likedBy: Meteor.userId(),
            },
          });
        } else {
          Collection.update({
            _id: documentId,
          }, {
            $addToSet: {
              likedBy: Meteor.userId(),
            },
          });
        }

        return !userAlreadyLiked ? 'Liked' : 'Not liked';
      },
    });
```


```javascript
Template.likesButton.events({

  'click #likesButton': function (event, templateInstance) {
    event.preventDefault();

    const collectionName = templateInstance.data.collectionName;

    const documentId = templateInstance.data.doc._id;

    Meteor.call('likeDocument', collectionName, documentId, (error) => {
      if (error) {
        console.log('error', error);
      }
    });
  },

});
```

```javascript

    Meteor.publishComposite('studentQuestionsForSchedule', function pubStudentQuestionsForSchedule(scheduleId) {
      check(scheduleId, String);

      if (!this.userId) {
        return this.ready();
      }

      const user = Users.findOne({ _id: this.userId });

      if (!user) {
        return this.ready();
      }
      const query = {
        schedule_id: scheduleId,
        userType: 'student',
      };
      if (!user.isLecturer(scheduleId)) {
        query.$or = [{
          visibleToOthers: {
            $ne: false,
          },
        }, {
          user_id: this.userId,
        }];
      }

      return {
        find() {
          return Questions.find(query);
        },
        children: [{
          find(question) {
            return Users.find(question.user_id);
          },
        }],
      };
    });
````

**Comment by theodorDiaconu** [comment]
@nadeemja thank you, will investigate.

**Comment by theodorDiaconu** [comment]
@nadeemja is your method optimistic ? do you execute it on client as well ?

**Comment by nadeemja** [comment]
No, but I'll try with optimistic ui too :)

**Comment by theodorDiaconu** [comment]
The issue is with publishComposite, I am now writing tests, because multiple people have the same problems.

**Comment by theodorDiaconu** [comment]
This issue has been addressed and will be fixed in 1.1.7
